src/mesh/difops.cxx

```diff
@@ -424,12 +379,15 @@ const Field3D Grad_par_LtoC(const Field3D &var) {
   } else {
     // No yup/ydown field, so transform to field aligned
 
-    Field3D var_fa = var.getMesh()->toFieldAligned(var);
+    Field3D var_fa = fieldmesh->toFieldAligned(var);
+    Field3D result_fa(fieldmesh);
+    result_fa.allocate();
 
     for(auto &i : result.region(RGN_NOBNDRY)) {
-      result[i] = (var_fa[i.yp()] - var_fa[i]) / (metric->dy[i]*sqrt(metric->g_22[i]));
+      result_fa[i] = (var_fa[i.yp()] - var_fa[i]) / (metric->dy[i]*sqrt(metric->g_22[i]));
```

Comment:
I don't think this is required, although it probably doesn't hurt beyond memory usage.

Reply:
Agreed it's not required. I thought it would be clearer to keep field-aligned and non-field-aligned fields separate, but am happy to revert back if people think it would be better. You make a good point about the memory usage, I'm inclined now to change it back to `result = mesh->fromFieldAligned(result);` style.